File: app/src/main/java/paul/barthuel/go4lunch/ui/notification/NotificationWorker.java
package paul.barthuel.go4lunch.ui.notification;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import com.google.firebase.auth.FirebaseAuth;

import java.io.IOException;
import java.util.List;

import paul.barthuel.go4lunch.data.firestore.restaurant.RestaurantRepository;
import paul.barthuel.go4lunch.data.firestore.restaurant.dto.Uid;
import paul.barthuel.go4lunch.data.firestore.user.UserRepository;
import paul.barthuel.go4lunch.data.firestore.user.dto.TodayUser;
import paul.barthuel.go4lunch.data.model.detail.Detail;
import paul.barthuel.go4lunch.data.retrofit.GooglePlacesAPI;
import paul.barthuel.go4lunch.data.retrofit.RetrofitService;
import retrofit2.Response;

public class NotificationWorker extends Worker {

    public NotificationWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
        super(context, workerParams);
    }

    @NonNull
    @Override
    public Result doWork() {

        NotificationHelper notificationHelper = new NotificationHelper(getApplicationContext());

        UserRepository userRepository = new UserRepository();
        //PlaceDetailRepository placeDetailRepository = new PlaceDetailRepository();
        RestaurantRepository restaurantRepository = new RestaurantRepository();

        if(FirebaseAuth.getInstance().getCurrentUser() != null) {
            if (userRepository.getTodayUser(FirebaseAuth.getInstance().getCurrentUser().getUid()) != null) {
                TodayUser todayUser = userRepository
                        .getTodayUserSync(FirebaseAuth
                                .getInstance()
                                .getCurrentUser()
                                .getUid());
                if (todayUser != null) {
                    GooglePlacesAPI service  = RetrofitService.getInstance().getGooglePlacesAPI();
                    try {
                        Response<Detail> detail = service.getDetailSearch(todayUser.getPlaceId(),
                                "formatted_address,name",
                                "AIzaSyDf9lQFMPnggxP8jYVT8NvGxmSQjuhNrNs").execute();
                        Detail detailSync = detail.body();

                        List<Uid> uids = restaurantRepository.getUidsSync(todayUser.getPlaceId());
                        StringBuilder allWorkmatesNames = new StringBuilder();
                        if (uids != null) {
                            for (Uid uid : uids) {
                                if(!uid.getUid().equals(todayUser.getUserId())) {
                                    allWorkmatesNames.append(uid.getWorkmateName());
                                }
                            }
                        }

                        if (detailSync != null) {
                            notificationHelper.displayNotification(detailSync.getResultDetail().getName() +
                                    " " +
                                    detailSync.getResultDetail().getFormattedAddress() +
                                    " " +
                                    allWorkmatesNames);
                        }

                    } catch (IOException e) {
                        e.printStackTrace();
                        Log.i("call", "fail");
                        return Result.failure();
                    }
                }
            }
        }

        Log.i("call", "succes");
        return Result.success();
    }
}

File: app/src/main/java/paul/barthuel/go4lunch/ui/notification/NotificationWorker.kt
package paul.barthuel.go4lunch.ui.notification

import android.content.Context
import android.util.Log
import androidx.work.Worker
import androidx.work.WorkerParameters
import com.google.firebase.auth.FirebaseAuth
import paul.barthuel.go4lunch.data.firestore.restaurant.RestaurantRepository
import paul.barthuel.go4lunch.data.firestore.user.UserRepository
import paul.barthuel.go4lunch.data.retrofit.RetrofitService
import java.io.IOException

class NotificationWorker(context: Context, workerParams: WorkerParameters) : Worker(context, workerParams) {
    override fun doWork(): Result {
        val notificationHelper = NotificationHelper(applicationContext)
        val userRepository = UserRepository()
        //PlaceDetailRepository placeDetailRepository = new PlaceDetailRepository();
        val restaurantRepository = RestaurantRepository()
        if (FirebaseAuth.getInstance().currentUser != null) {
            if (userRepository.getTodayUser(FirebaseAuth.getInstance().currentUser!!.uid) != null) {
                val todayUser = userRepository
                        .getTodayUserSync(FirebaseAuth
                                .getInstance()
                                .currentUser
                                ?.uid)
                if (todayUser != null) {
                    val service = RetrofitService.instance?.googlePlacesAPI
                    try {
                        val detail = service?.getDetailSearch(todayUser.placeId,
                                "formatted_address,name",
                                "AIzaSyDf9lQFMPnggxP8jYVT8NvGxmSQjuhNrNs")?.execute()
                        val detailSync = detail?.body()
                        val uids = restaurantRepository.getUidsSync(todayUser.placeId)
                        val allWorkmatesNames = StringBuilder()
                        if (uids != null) {
                            for (uid in uids) {
                                if (uid.uid != todayUser.userId) {
                                    allWorkmatesNames.append(uid.workmateName)
                                }
                            }
                        }
                        if (detailSync != null) {
                            notificationHelper.displayNotification(detailSync.resultDetail.name +
                                    " " +
                                    detailSync.resultDetail.formattedAddress +
                                    " " +
                                    allWorkmatesNames)
                        }
                    } catch (e: IOException) {
                        e.printStackTrace()
                        Log.i("call", "fail")
                        return Result.failure()
                    }
                }
            }
        }
        Log.i("call", "succes")
        return Result.success()
    }
}